### PR TITLE
feature/add pagination and redis

### PR DESCRIPTION
**Pull request by borovHokage**

### CODE REVIEW

File: src/FilterService.Broker/Requests/PositionService.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    public async Task<List<PositionFilteredData>> GetPositionFilterDataAsync(List<Guid> positionsIds, List<string> errors)
+    {
+      if (positionsIds is null || !positionsIds.Any())
+      {
+        return null;
+      }
+
+      List<PositionFilteredData> positionsData = await _globalCache.GetAsync<List<PositionFilteredData>>(Cache.Positions, positionsIds.GetRedisCacheHashCode());
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/PositionService.cs:44]: add empty line

File: src/FilterService.Broker/Requests/RoleService.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public async Task<List<RoleFilteredData>> GetRolesFilterDataAsync(List<Guid> roleIds, List<string> errors)
+    {
+      if (roleIds is null || !roleIds.Any())
+      {
+        return null;
+      }
+
+      List<RoleFilteredData> rolesData = await _globalCache.GetAsync<List<RoleFilteredData>>(Cache.Rights, roleIds.GetRedisCacheHashCode());
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/RoleService.cs:41]: add empty line

File: src/FilterService.Broker/Requests/PositionService.cs (hunk 1 above)
Comment (borovHokage) [reply src/FilterService.Broker/Requests/PositionService.cs:44]: done

File: src/FilterService.Broker/Requests/OfficeService.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public async Task<List<OfficeFilteredData>> GetOfficeFilterDataAsync(List<Guid> officesIds, List<string> errors)
+    {
+      if (officesIds is null || !officesIds.Any())
+      {
+        return null;
+      }
+
+      List<OfficeFilteredData> officesData = await _globalCache.GetAsync<List<OfficeFilteredData>>(Cache.Offices, officesIds.GetRedisCacheHashCode());
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/OfficeService.cs:41]: add empty line

File: src/FilterService.Broker/Requests/DepartmentService.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+      if (departmentsData is null)
+      {
+        departmentsData =
+          (await RequestHandler.ProcessRequest<IFilterDepartmentsRequest, IFilterDepartmentsResponse>(
+          _rcGetDepartments,
+          IFilterDepartmentsRequest.CreateObj(departmentsIds),
+          errors,
+          _logger))
+        .Departments;
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/DepartmentService.cs]: check if null

File: src/FilterService.Broker/Requests/DepartmentService.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+      if (departmentsData is null)
+      {
+        departmentsData =
+          (await RequestHandler.ProcessRequest<IGetDepartmentsRequest, IGetDepartmentsResponse>(
+          _rcGetDepartmentsData,
+          IGetDepartmentsRequest.CreateObj(usersIds),
+          errors,
+          _logger))
+        .Departments;
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/DepartmentService.cs]: check if null

File: src/FilterService.Broker/Requests/ImageService.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        return null;
+      }
+
+      return
+        (await RequestHandler.ProcessRequest<IGetImagesRequest, IGetImagesResponse>(
+          _rcGetImages,
+          IGetImagesRequest.CreateObj(usersImageIds, ImageSource.User),
+          errors,
+          _logger))
+        .ImagesData;
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/ImageService.cs]: check if null

File: src/FilterService.Broker/Requests/DepartmentService.cs [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+          (await RequestHandler.ProcessRequest<IFilterDepartmentsRequest, IFilterDepartmentsResponse>(
+            _rcGetDepartments,
+            IFilterDepartmentsRequest.CreateObj(departmentsIds),
+            errors,
+            _logger))
+          ?.Departments;
+      }
+      if (departmentsData is null)
+      {
+        errors.Add("Cannot get Departments");
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/DepartmentService.cs]: "Can not filter by department."

File: src/FilterService.Broker/Requests/DepartmentService.cs [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+      {
+        departmentsData =
+          (await RequestHandler.ProcessRequest<IGetDepartmentsRequest, IGetDepartmentsResponse>(
+            _rcGetDepartmentsData,
+            IGetDepartmentsRequest.CreateObj(usersIds),
+            errors,
+            _logger))
+          ?.Departments;
+      }
+      if (departmentsData is null)
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/DepartmentService.cs]: remove

File: src/FilterService.Broker/Requests/ImageService.cs [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+      List<ImageData> imagesData =
+        (await RequestHandler.ProcessRequest<IGetImagesRequest, IGetImagesResponse>(
+          _rcGetImages,
+          IGetImagesRequest.CreateObj(usersImageIds, ImageSource.User),
+          errors,
+          _logger))
+        ?.ImagesData;
+
+      if (imagesData is null)
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/ImageService.cs]: remove

File: src/FilterService.Broker/Requests/OfficeService.cs [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+          (await RequestHandler.ProcessRequest<IFilterOfficesRequest, IFilterOfficesResponse>(
+            _rcGetOffices,
+            IFilterOfficesRequest.CreateObj(officesIds),
+            errors,
+            _logger))
+          ?.Offices;
+      }
+      if (officesData is null)
+      {
+        errors.Add("Cannot get Offices");
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/OfficeService.cs]: "Can not filter by office."

File: src/FilterService.Broker/Requests/PositionService.cs [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+          (await RequestHandler.ProcessRequest<IFilterPositionsRequest, IFilterPositionsResponse>(
+            _rcGetPosotions,
+            IFilterPositionsRequest.CreateObj(positionsIds),
+            errors,
+            _logger))
+          ?.Positions;
+      }
+      if (positionsData is null)
+      {
+        errors.Add("Cannot get Positions");
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/PositionService.cs]: "Can not filter by position."

File: src/FilterService.Broker/Requests/PositionService.cs [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+      {
+        positionsData =
+          (await RequestHandler.ProcessRequest<IGetPositionsRequest, IGetPositionsResponse>(
+            _rcGetPositionsData,
+            IGetPositionsRequest.CreateObj(usersIds),
+            errors,
+            _logger))
+          ?.Positions;
+      }
+      if (positionsData is null)
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/PositionService.cs]: remove

File: src/FilterService.Broker/Requests/RoleService.cs [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+          (await RequestHandler.ProcessRequest<IFilterRolesRequest, IFilterRolesResponse>(
+            _rcGetRoles,
+            IFilterRolesRequest.CreateObj(roleIds),
+            errors,
+            _logger))
+          ?.Roles;
+      }
+      if (rolesData is null)
+      {
+        errors.Add("Cannot get Roles");
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/RoleService.cs]: "Can not filter by role."

File: src/FilterService.Broker/Requests/UserService.cs [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+      {
+        usersData =
+          (await RequestHandler.ProcessRequest<IGetUsersDataRequest, IGetUsersDataResponse>(
+            _rcGetUsers,
+            IGetUsersDataRequest.CreateObj(usersIds, value.SkipCount, value.TakeCount),
+            errors,
+            _logger))
+          ?.UsersData;
+      }
+      if (usersData is null)
```
Comment (MariyaSin) [inline src/FilterService.Broker/Requests/UserService.cs]: remove

File: src/FilterService.Business/Commands/User/FilterUsersCommand..cs [hunk 15]
```diff
@@ -34,219 +24,115 @@ namespace LT.DigitalOffice.FilterService.Business.Commands.User
...
-    private async Task<List<DepartmentData>> GetDepartmentsDataAsync(
-      List<Guid> usersIds,
+    private async Task<List<UserData>> GetUsersDataAsync(
+      PaginationValues value,
       List<string> errors)
     {
-      return (await RequestHandler.ProcessRequest<IGetDepartmentsRequest, IGetDepartmentsResponse>(
-          _rcGetDepartmentsData,
-          IGetPositionsRequest.CreateObj(usersIds),
+      List<UserData> usersData =
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/FilterUsersCommand..cs]: use broker.requests.UserService

File: src/FilterService.Business/Commands/User/FilterUsersCommand..cs [hunk 16]
```diff
@@ -34,219 +24,115 @@ namespace LT.DigitalOffice.FilterService.Business.Commands.User
...
       _positionInfoMapper = positionInfoMapper;
       _departmentInfoMapper = departmentInfoMapper;
       _officeInfoMapper = officeInfoMapper;
+      _officeService = officeService;
+      _departmentService = departmentService;
+      _imageService = imageService;
+      _positionService = positionService;
+      _roleService = roleService;
+      _userService = userService;
       _logger = logger;
```
Comment (MariyaSin) [inline src/FilterService.Business/Commands/User/FilterUsersCommand..cs]: do you use logger here?

**Review by MariyaSin — APPROVED** [review]

**Review by ekaterinaryzhenkova — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**